### PR TITLE
reduce the number of default targets

### PR DESCRIPTION
**Pull request by illsilin**
This is a fresh pull request with the new options to build individual targets, as suggested by Adam.

### CODE REVIEW

**Review by junliume — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -253,9 +253,27 @@ rocm_package_setup_component(tests
...
+    if(WITH_EXAMPLES)
+        add_subdirectory(example)
+    endif()
+    if(WITH_TESTS)
+        add_subdirectory(test)
+    endif()
+    if(WITH_PROFILER)
+        add_subdirectory(profiler)
+    endif()
+endif()
```
Comment (aosewski) [inline CMakeLists.txt:276]: Hey, what about following: 
```suggestion
if(BUILD_ALL)
    set(WITH_EXAMPLES ON)
    set(WITH_TESTS ON)
    set(WITH_PROFILER ON)
endif()

if(WITH_EXAMPLES)
    add_subdirectory(example)
endif()
if(WITH_TESTS)
    add_subdirectory(test)
endif()
if(WITH_PROFILER)
    add_subdirectory(profiler)
endif()
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -253,9 +253,27 @@ rocm_package_setup_component(tests
...
 
 add_subdirectory(library)
-add_subdirectory(example)
-add_subdirectory(test)
-add_subdirectory(profiler)
+
+option(BUILD_ALL, "Whether to build all targets, such as examples, tests, and profiler." OFF)
+option(WITH_EXAMPLES, "Whether to build examples." OFF)
+option(WITH_TESTS, "Whether to build tests." OFF)
+option(WITH_PROFILER, "Whether to build profiler." OFF)
```
Comment (aosewski) [inline CMakeLists.txt:260]: Can we place all cmake options in one place? There is already one for INT4 in this file some lines above.

**Review by aosewski — APPROVED** [review]
Looks good. Just small refactoring to consider.

File: CMakeLists.txt (hunk 1 above)
Comment (illsilin) [reply CMakeLists.txt:276]: Sure, we can do it this way.

File: CMakeLists.txt (hunk 2 above)
Comment (illsilin) [reply CMakeLists.txt]: OK

**Comment by illsilin** [comment]
After discussion with Chao, the decision was to leave the default behavior of CK unchanged, and instead to add a new custom target "instances" that will build all instances that may be required by MIOpen or other libraries.

**Review by asroy — APPROVED** [review]

**Pull request merged**